### DISCUSSION TITLE
Avatar field ratio on details

### DISCUSSION THREAD

**Discussion by mrleblanc101** [laravel/nova-issues#4930] [Ideas & Feature Requests] [answered]
- Laravel Version: 9.34.0
- Nova Version: 4.15.2
- PHP Version: 8.1
- Database Driver & Version: mariadb 10.4
- Operating System and Version: macOS 12.6
- Browser type and version: Chrome 105

### Description:
The Avatar field when using `->rounded` is not actually round on Details. 
It default to the image ratio and apply a border-radius.
http://admin.local/resources/people/1/ 
![Capture d’écran, le 2022-10-14 à 14 51 14](https://user-images.githubusercontent.com/17392251/195920097-5c602ab9-c873-46ec-8211-d103b05ff218.png)

But it is round on Forms with a static 1:1 ratio.
The behavior should be the same on both view.
http://admin.local/resources/people/1/edit/ 
![Capture d’écran, le 2022-10-14 à 14 53 24](https://user-images.githubusercontent.com/17392251/195920191-5f7de4f7-9bc4-40d1-8fa5-900f2ad350a3.png)

**Comment by mrleblanc101** [comment]
Expected display on details page:
![Capture d’écran, le 2022-10-14 à 14 58 15](https://user-images.githubusercontent.com/17392251/195920998-a0347c9e-4554-42b4-9708-b4d45f044c68.png)
Need to add a wrapper with these class: `bg-gray-50 relative aspect-square flex items-center justify-center border-2 border-gray-200 dark:border-gray-700 overflow-hidden rounded-full` and move the `style="max-width: 104px"` to this wrapper.

**Comment by mrleblanc101** [comment]
Same issue when using `->squared()`

**Comment by kichetof** [comment]
Same here! Seem to happened from 4.15 version

**Comment by mrleblanc101** [comment]
Why was this concerted to a discussion ? It's clearly a regression @crynobone

**Reply by crynobone** [reply]
Behaviour on detail page hasn't change between 4.14 vs 4.15

**Reply by mrleblanc101** [reply]
It's still a bug, tho ?

**Reply by crynobone** [reply]
All implemented feature request solves some sort of bug and I believe what's currently available is the expected behavior and not any type of regression (as mentioned above). Therefore, we would need to carefully consider any change.

To accept a regression would mean the form is wrong and we need to remove the `1:1` ratio.

**Reply by mrleblanc101** [reply]
Alright. 
But I do think the 1:1 ratio is the correct behavior for avatar to display avatar. 
The 1:1 ratio is also used when using Searchable.
<img width="630" alt="avatar-poster 00f6ebb8" src="https://user-images.githubusercontent.com/17392251/196091301-103eee03-b2be-4458-a7d1-bcc199f68803.png">

And it make sense to use Image field instead if you dont want to display the image in a circle container.
So I would suggest correcting the index/details of the Avatar field.
This would only affect the display of the image, as the avatar field doesn't force a 1:1 ratio on upload.
Site using the Avatar data could still display the image using the original ratio or use contain/cover fill.

**Reply by mrleblanc101** [reply]
From what I understand the Avatar field is a wrapper for the Image field which only purpose is to be displayed in Searchable relation and being display in a Circle container (because that's how Avatar are normally displayed).

**Reply by kichetof** [reply]
Maybe add a new helper `circle` to force it to be completely round as before.

It seem that from the updated `File` field introduced in 4.15, the ratio is used to display avatar and impact the visual

**Comment by kichetof** [comment]
Please find the difference between 4.16.1 (same with 4.15) and 4.14.0: _(same code, `Avatar::make('Avatar')`, only changed Nova version)_

**Nova 4.16.1**
Index avatar should be fully-rounded, regardless of its initial ratio, according to the doc: `Additionally, you may use the rounded method to display its thumbnails with fully-rounded edges`
![Nova 4.16.1 index](https://i.postimg.cc/t4Q4LPj9/Capture-d-e-cran-2022-10-17-a-09-41-33.png)

**Nova 4.14.0**
Index avatar is fully-rounded, as expected
![Nova 4.14 index](https://i.postimg.cc/W1G1j3Ng/Capture-d-e-cran-2022-10-17-a-09-45-01.png)

**Comment by davidhemphill** [comment, marked as answer]
Made some tweaks to the `Avatar` and `Image` fields. In the next version, an `aspect` method will allow you to set the image's aspect ratio. Nova will not assume the aspect ratio by default, meaning your `Avatar` fields will not be presumed to be square.

```php
use Laravel\Nova\Fields\Image;

Image::make('Image')
    ->indexWidth(70)
    ->detailWidth(300)
    ->rounded()
    ->aspect(Image::ASPECT_SQUARE)
```

The options available are:

- ASPECT_SQUARE
- ASPECT_AUTO

I've also added `indexWidth` and `detailWidth` methods to allow you to specify separate sizes for each view. It's important to note, `maxWidth` will override both of those for compatibility purposes, but is now `null` by default.

**Comment by kichetof** [comment]
@davidhemphill thanks for this update!

It could be great to define aspect on both view like width: `indexAspect` and `detailAspect`.
When the height of the image is too big, the height of the row become very high. Maybe add a possibility to set `maxHeight`?

To reproduce it, set an avatar with a ratio 3:2 or 3:1

_ASPECT_VIDEO seems to be not present on 4.18.0 release, only ASPECT_SQUARE and ASPECT_AUTO_
